lib/grape/endpoint.rb

```diff
@@ -339,10 +338,9 @@ def lazy_initialize!
     end
 
     def run_validators(validators, request)
-      validation_errors = []
-
-      validators.each do |validator|
+      validators.each_with_object(validation_errors = []) do |validator|
         begin
+          validator.validators = validators
```

Comment:
This introduces state during validation into the validator proper, I don't think that's OK since these are singleton objects.

Or do we do this elsewhere?